kevm_pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -98,24 +96,20 @@ def gen_claims_for_contract(empty_config: KInner, contract_name: str) -> List[KC
         'ACCOUNTS_CELL': KApply('_AccountCellMap_', [account_cell, KVariable('ACCOUNTS')]),
     }
     final_subst = {'K_CELL': KSequence([KEVM.halt(), KVariable('CONTINUATION')])}
-    init_term = CTerm(substitute(empty_config, init_subst))
-    final_term = CTerm(abstract_cell_vars(substitute(empty_config, final_subst)))
-    claim, _ = build_rule(contract_name.lower(), init_term, final_term, claim=True)
-    assert isinstance(claim, KClaim)
-    return [claim]
+    init_term = substitute(empty_config, init_subst)
+    init_terms = [(contract_name.lower(), init_term)]
+    if calldata_cells:
+        init_terms = [(f'{contract_name.lower()}-{i}', substitute(init_term, {'CALLDATA_CELL': cd})) for i, cd in enumerate(calldata_cells)]
+    final_term = abstract_cell_vars(substitute(empty_config, final_subst))
+    claims: List[KClaim] = []
+    for claim_id, i_term in init_terms:
+        claim, _ = build_rule(claim_id, CTerm(i_term), CTerm(final_term), claim=True)
+        assert isinstance(claim, KClaim)
```

Comment:
We should have a function `build_claim` in `pyk` that wraps this logic.

Reply:
Added here: https://github.com/runtimeverification/ksummarize/issues/82